interface/src/Application.cpp

```diff
@@ -7530,6 +7532,15 @@ void Application::registerScriptEngineWithApplicationServices(const ScriptEngine
     scriptEngine->registerGlobalObject("HifiAbout", AboutUtil::getInstance());  // Deprecated.
     scriptEngine->registerGlobalObject("ResourceRequestObserver", DependencyManager::get<ResourceRequestObserver>().data());
 
+    // This is obviously wrong -- FIXME!
+    scriptEngine->registerValue("Bucket.Public", static_cast<int>(ExternalResource::Bucket::Public));
+    scriptEngine->registerValue("Bucket.Content", static_cast<int>(ExternalResource::Bucket::Content));
+    scriptEngine->registerValue("Bucket.MPAssets", static_cast<int>(ExternalResource::Bucket::MPAssets));
+    scriptEngine->registerValue("Bucket.Assets", static_cast<int>(ExternalResource::Bucket::Assets));
```

Comment:
This is bad. How do I do it better?

Reply:
what if you exposed these as QPROPERTIES on the ExternalResource class so that you can do things like ExternalResource.Public?  other scripting interfaces have examples, like this: https://github.com/kasenvr/project-athena/blob/master/interface/src/raypick/PickScriptingInterface.h

Reply:
I poked at the problem a bit more and came up with what's in the commit I just added -- how does that look? It seems fairly similar to what you linked.

Reply:
that’s definitely cleaner!  I still think that since the Bucket enum is only used by the ExternalResource interface, the enum should be a property of ExternalResource.  that way we don’t clutter up the scripting API with two different namespaces